src/bin/strategy.rs

```diff
@@ -0,0 +1,195 @@
+
+    let backend = Backend::new(sdk.as_ref(), insight.clone(), config.clone()).await;
+
+    // Create wallet if not initialized
+    if backend.state().loaded_wallet.lock().await.is_none() {
+        let Some(private_key) = config.wallet_private_key else {
+            panic!("Wallet not initialized and no private key provided");
+        };
+
+        tracing::info!("Wallet not initialized, creating new wallet with configured private key");
+
+        backend
+            .run_task(Task::Wallet(WalletTask::AddByPrivateKey(
+                private_key.clone(),
+            )))
+            .await;
+    }
+
+    // Refresh wallet core balance
+    backend.run_task(Task::Wallet(WalletTask::Refresh)).await;
+
+    // Return if there's no loaded identity and the --dash flag is not set
+    if backend.state().loaded_identity.lock().await.is_none() && args.dash.is_none() {
+        panic!("There's no loaded identity and the --dash argument is not passed");
+    }
+    
+    // Add loaded identity to known identities if it's not already there
+    // And set selected_strategy to None
+    {
+        let state = backend.state();
+        let loaded_identity = state.loaded_identity.lock().await;
+        let mut selected_strategy = state.selected_strategy.lock().await;
+        let mut known_identities = state.known_identities.lock().await;
+
+        if let Some(loaded_identity) = loaded_identity.as_ref() {
+            known_identities
+                .entry(loaded_identity.id())
+                .or_insert_with(|| loaded_identity.clone());
+        }
+
+        *selected_strategy = None;
+    }
+    
+    // Handle CLI commands
+    if let Some(start_dash) = args.dash {
+        // Register identity with `dash` balance if there is none yet
+        if backend.state().loaded_identity.lock().await.is_none() {
+            let amount = start_dash * 100000000; // duffs to go into asset lock transaction
+
+            tracing::info!(
+                "Identity not registered, registering new identity with {} Dash",
+                start_dash
+            );
+
+            backend
+                .run_task(Task::Identity(IdentityTask::RegisterIdentity(amount)))
+                .await;
+        // Else, if there is a loaded identity, if the balance is less than start_dash, top it up
+        } else {
+            backend.run_task(Task::Wallet(WalletTask::Refresh)).await;
+            backend
+                .run_task(Task::Identity(IdentityTask::Refresh))
+                .await;
+
+            let balance = backend
+                .state()
+                .loaded_identity
+                .lock()
+                .await
+                .as_ref()
+                .unwrap()
+                .balance();
+
+            tracing::info!(
+                "Platform wallet has {} Dash",
+                balance as f64 / 100000000000.0
+            );
+
+            if balance < start_dash * 100000000000 {
```

Comment:
i think these constants and stuff should be made as a separate type;

we have `IdentityBalance` with a constructor method that builds a type from credits, so we have at least some type safety

what I suggest is to have something similar or move and rename `IdentityBalance` to `DashBalance` and use it both for storing balance data and formatting so we won't compare it to plain numbers; these zeroes are confusing